doc/src/location_tracking.md

```diff
@@ -0,0 +1,25 @@
+# Location Tracking
+
+Sometimes it may be helpful to know where in the input stream a particular
+token was encountered. For example, this can be useful when generating
+user-visible messages that reference specific points in the input.
+
+This is achieved via the location tracking macros, `@L` and `@R`.  `@L` looks
+ahead to the token immediately to its right, and binds the the location (in
+bytes) in the input stream where that token starts.  Similarly, `@R` binds the
```

Comment:
Nit: should we mention the exact type instead of just "bytes"(which seems to be usize)?